tests/test_api_render.py

```diff
@@ -182,6 +182,28 @@ def test_cross_info_index_platform(testing_config):
     assert metadata.config.host_platform == info_index["platform"]
 
 
+def test_noarch_with_platform_deps(testing_workdir, testing_config):
+    recipe_path = os.path.join(metadata_dir, "_noarch_with_platform_deps")
+    build_ids = {}
+    for subdir in ["linux-64", "linux-aarch64", "linux-ppc64le", "osx-64", "win-64"]:
+        m = api.render(recipe_path, config=testing_config, subdir=subdir)[0][0]
+        build_ids[subdir] = m.build_id()
```

Comment:
[`subdir` is a read-only property](https://github.com/conda/conda-build/blob/313fb69261135db7db0ccc7a6fa26e6d3fb426ba/conda_build/config.py#L417) so it can't be set on the `Config` object. We can set `platform` and `arch` though, so that's the fix for the failing test.

```suggestion
    for subdir_ in ["linux-64", "linux-aarch64", "linux-ppc64le", "osx-64", "win-64"]:
        platform, arch = subdir_.split("-")
        m = api.render(recipe_path, config=testing_config, platform=platform, arch=arch)[0][0]
        build_ids[subdir_] = m.build_id()
```

Reply:
(I think 😬)